core/src/smartcontracts/wasm.rs

```diff
@@ -539,11 +562,12 @@ impl<'wrld> Runtime<'wrld> {
         instruction_judge: InstructionJudgeArc,
         query_judge: QueryJudgeArc,
     ) -> Result<(), Error> {
-        let state = State::new(wsv, account_id.clone(), self.config).with_validator(
-            max_instruction_count,
-            instruction_judge,
-            query_judge,
-        );
+        let curr_time_millis = current_time()
+            .as_millis()
+            .try_into()
+            .expect("Timestamp millis should fit into u64");
```

Comment:
This might be controversial, but why can't this just be `u128`? It solves a problem that we're unlikely to encounter, but still an unnecessary panic.

Reply:
We already have `Numeric::BigQuantity` which is `u128` and crossing the FFI boundary, so why not make everything work without conversions?

Reply:
This cannot fail(for ~500million years) unless system clock is tampered with. I made this choice because `u64` is natively supported by `WebAssembly`, for `u128` it gets more hairy. I don't think FFI library can be used here because this is communication between wasm module and wasm runtime

a counter argument - do you think trigger can leverage the subsecond precision?

Reply:
> a counter argument - do you think trigger can leverage the subsecond precision?

Reply:
I think we should be extra careful with passing current time inside trigger. 
Suppose that peer try to reconnect to the network network it will commits blocks at different time that they are committed originally and it could lead to divergence. 
Maybe we can pass inside block related timestamp?